src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/readytolist/ReadyToListAboutToSubmitHandler.java

```diff
@@ -25,24 +44,48 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
     }
 
     @Override
-    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
-                                                          String userAuthorisation) {
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
         if (!canHandle(callbackType, callback)) {
             throw new IllegalStateException("Cannot handle callback.");
         }
+
         final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
-        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData(); //Maybe handle in each method instead of HEAD
 
-        PreSubmitCallbackResponse<SscsCaseData> callbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+        if(schedulingAndListingFeature){
+            boolean listAssist = checkIfListAssist();
+            if(listAssist){
+                //Build response to be enium value and case ID
+                JSONObject buildJson = new JSONObject();
+                buildJson.put("STATE", STATE.HEARING_CREATED);
+                buildJson.put("CaseID", sscsCaseData.getCcdCaseId());
+                PreSubmitCallbackResponse<SscsCaseData> callbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData); //Placeholder
+                return callbackResponse;
+            }else{
+                return handleCallbackResponse(sscsCaseData); //Return back to old method
+            }
+        }else{
+            return handleCallbackResponse(sscsCaseData); //Return back to old method
+        }
+    }
 
-        log.info(String.format("createdInGapsFrom is %s for caseId %s",
-                sscsCaseData.getCreatedInGapsFrom(), sscsCaseData.getCcdCaseId()));
+    private boolean checkIfListAssist(){
+        //Read file rpc-data from JSON
+        //Map out JSON object
+        //filter and then return if true or false below on listAssist
+        //boolean isListAssist = object.getListAssist : true : false
```

Comment:
Jackson lets you do this in a couple of lines. All you'll need to do is do a Map lookup using the case data location as the key.